scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -89,9 +97,29 @@ private static TaskState getInitialStateForStreams(Map<TaskId, SortedSet<StreamI
      * This includes fetching saved state of each task or creating a new initial
      * state for tasks that haven't run successfully before.
      */
-    private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams) throws ExecutionException, InterruptedException {
-        return new TaskActionsQueue(createTasksWithState(groupedStreams)
-                .map(task -> TaskAction.createFirstAction(workerConfiguration, task)).collect(Collectors.toSet()));
+    private Collection<TaskAction> queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams)
+            throws ExecutionException, InterruptedException {
+        return createTasksWithState(groupedStreams).map(task -> TaskAction.createFirstAction(workerConfiguration, task))
+                .collect(Collectors.toSet());
+    }
+
+    private boolean shouldStop() {
```

Comment:
What's the purpose of this method? Can't shouldStop be used directly? The method is private so all the callers have access to the field as well, no?

Reply:
Of course it can. I am just old enough to use private wrapper methods for these things, just in case one ever wants to move from a simple check var to for example an expression. Abstraction.

Reply:
Sounds to me like a not very useful abstraction when it's used only inside a single class :) but ok.